### DISCUSSION TITLE
Meeting Minutes from January 16, 2024

### DISCUSSION THREAD

**Discussion by dezidizon** [CVWRSM/cvwrsm#39] [Meeting Minutes]
**Open Action Items:**
- Coordinate with Hao Xie for manual steps in creating WRIMS installer and patch artifacts. This is tracked as #29.
- Apply copyright files (license and source) in the wrims repository using Eclipse public License (EPL) version 2 or 3. This is tracked as #16.
- RMA has sent a DSS global variable flag for Hao Xie to use in order to force WRIMS to write DSS6 files. RMA will help diagnosing the issues as this is not working correctly. This is tracked as #31.
- Remove vista libraries from WRIMS. This is tracked as #33.
- Document descriptions for each folder in WRIMS. This is tracked as #36.

**Discussions:**
-  Hao asked whether Eclipse can still be used as the IDE for working with the updated repository for WRIMS.
    - Peter provided clarification regarding having the source code separated by language type - antlr and java. Antlr is a templating language the preprocesses java code.
    - Eclipse can work with the gradle build. Once the gradle build is run, it is run as part of the build process in eclipse.
- Hao approved creating the new branch of CentralValleyModeling/wrims/feature/wrims-devops which includes the updates for the WRIMS DevOps effort.
    - Tom's effort in updating the directory structure for the wrims repository to work with gradle is tracked in this branch.
    - Very strict control of how code is merged into this new branch will be implemented. No direct commits will be permitted. Commits must come through in peer approval. Conversations must be resolved before a pull request is merged into the wrims-devops branch.
    - A set of GitHub actions will be brought in to compose the jars and eventually assemble into wrims.
- Raymond Hoang requested insight regarding branch security under the CentralValleyModeling GitHub organization.
    - Raymond asked for recommendations for protecting the master branch for the wrims repository.
    - Peter recommends using the same standards as noted for the new branch of feature/wrims-devops.
    - Adam discussed adding static code analysis. This ensures that all code compiles before being able to merge. 
    - RMA will be happy to assist DWR.
- Tom walked through the documentation effort done for the wrims repository as part of the WRIMS DevOps effort. This is tracked as #38.
    - The documentation effort is managed in the same way that the building environment was discussed earlier.
    - A pull request was created to include the updates for the README files. This may be marked as a draft so that it is not yet merged until it is complete.
    - Discussion with Hao is important to inform us of the purpose of each folder in the wrims repository. Tom looked at directories and the dependencies among the code to figure out what may be the purpose. Ultimately, Hao's feedback inform the actual purpose for each folder.
    - Tom will continue to add pieces. Hao will reply to the pull request. Hao will provide review, corrections, and addition of content after Tom is done.
    - Hao noted that the only issue he saw in the documentation was content referring to using a batch file to create the java code.
- The next effort will focus on making sure that the plugin jars can be built in gradle.
    - There may be a challenge in assembling the application to put the framework together so that the plugins are plugged in effectively. The process has been dependent on the Eclipse IDE  so far.
- Hao confirmed to Tom that the ITextEditor and ITextEditorUI include generated code.
    - Tom has difficulty telling what the generated code is here. He will need to be able to generate code or run the code generation process as he did for antlr in the wrims2 build.

**Closed Action Items:**
- Yiwei Cheng will provide historical study to use for comparing results from existing WRIMSv2.jar and updated WRIMSv2.jar. This is tracked as #36.
- Provide access to Hao Xie to review the updates for the WRIMS DevOps work.

**Comment by dezidizon** [comment]
#41 was created to track the creation of the new branch of CentralValleyModeling/wrims/feature/wrims-devops. This will be tracked as an open action item for the next meeting minutes.

**Comment by rma-tom** [comment]
Task "Document descriptions for each folder in WRIMS" is tracked as issue #34 not 36
